src/sql/workbench/test/common/editor/query/gridTableState.test.ts

```diff
@@ -0,0 +1,62 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { GridTableState } from 'sql/workbench/common/editor/query/gridTableState';
+import * as assert from 'assert';
+import { isUndefined } from 'vs/base/common/types';
+import { Event } from 'vs/base/common/event';
+
+suite('Grid Table State', () => {
+	test('inital state is correct', () => {
+		const resultId = 0;
+		const batchId = 0;
+		const state = new GridTableState(resultId, batchId);
+
+		assert.equal(state.resultId, resultId);
+		assert.equal(state.batchId, batchId);
+		assert(isUndefined(state.canBeMaximized));
+		assert(isUndefined(state.maximized));
+		assert.equal(state.scrollPositionX, 0);
+		assert.equal(state.scrollPositionY, 0);
+		assert(isUndefined(state.columnSizes));
+		assert(isUndefined(state.selection));
+		assert(isUndefined(state.activeCell));
+	});
+
+	test('does set properties correctly', async () => {
+		const state = new GridTableState(0, 0);
+		let event = await new Promise<boolean>(resolve => {
+			Event.once(state.onCanBeMaximizedChange)(e => resolve(e));
+			state.canBeMaximized = true;
+		});
+
+		assert.equal(event, true);
+		assert.equal(state.canBeMaximized, true);
+
+		event = await new Promise<boolean>(resolve => {
+			Event.once(state.onCanBeMaximizedChange)(e => resolve(e));
+			state.canBeMaximized = false;
+		});
+
+		assert.equal(event, false);
+		assert.equal(state.canBeMaximized, false);
```

Comment:
Would be good to add a message to these saying what it was comparing - the failure isn't going to be enough information here alone to figure out what part was wrong.

Reply:
I'm not sure what the value of a message here would be, the message would be ~`expected can be maximized to be false` which is self evident by the assert I assume.

Reply:
But the assert failing is just going to say something like `AssertionError [ERR_ASSERTION]: false == true` which isn't terrible useful when you're doing multiple of these checks in the same test.

Reply:
And this would be for failures in automated runs mostly where you wouldn't be able to debug it. It's useful to be able to tell exactly what condition failed from the logs alone in case you can't repro it right away.